q2_longitudinal/tests/test_longitudinal.py

```diff
@@ -288,6 +310,149 @@ def test_nmit_missing_table_ids(self):
             nmit(table=self.table_taxa_fp, metadata=md,
                  individual_id_column='studyid')
 
+    def test_first_differences(self):
+        exp = pd.Series([0.08, 0.06, 0.07999999999999999, 0.12, 0.14,
+                         0.14999999999999997],
+                        index=['3', '4', '5', '9', '10', '11'],
+                        name='Difference')
+        exp.index.name = '#SampleID'
+        obs = first_differences(
+            metadata=qiime2.Metadata(md), state_column='Time',
+            individual_id_column='ind',
+            metric='Value', replicate_handling='drop')
+        pdt.assert_series_equal(obs.sort_index(), exp.sort_index())
+
+    # what if nothing changes between time points?
+    def test_first_differences_static(self):
+        exp = pd.Series([0., 0., 0., 0., 0., 0.],
+                        index=['3', '4', '5', '9', '10', '11'],
+                        name='Difference')
+        exp.index.name = '#SampleID'
+        obs = first_differences(
+            metadata=qiime2.Metadata(md_static), state_column='Time',
+            individual_id_column='ind',
+            metric='Value', replicate_handling='drop')
+        pdt.assert_series_equal(obs.sort_index(), exp.sort_index())
+
+    def test_first_differences_drop_duplicates(self):
+        obs = first_differences(
+            metadata=qiime2.Metadata(md_dup), state_column='Time',
+            individual_id_column='ind',
+            metric='Value', replicate_handling='random')
+        # The first diff of individual 2 is subject to random rep handling
+        mystery_number = obs.iloc[1]
+        self.assertTrue(mystery_number in [0.04999999999999999, 0.06])
```

Comment:
This check could fail on different platforms/architectures due to floating point error. Can you use [`self.assertAlmostEqual()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual) instead of exact equality?

Reply:
👍